scalameta/inputs/shared/src/main/scala/scala/meta/inputs/Input.scala

```diff
@@ -48,6 +48,12 @@ object Input {
     }
   }
 
+  @data class VirtualFile(path: AbsolutePath, contents: scala.Predef.String) extends Input {
+    override def location = s"<${path.absolute}>"
+    lazy val chars = contents.toArray
+    override def toString = s"""Input.VirtualFile(AbsolutePath("${path.relative}"), "${contents.take(10)}...")""""
```

Comment:
The `take(10)` part is a bit inconsistent with prettyprinting of `Input.String`. How about we avoid trimming here?